HOWTOCONTRIBUTE.rst

```diff
@@ -0,0 +1,231 @@
+`here <https://github.com/ralsina/rst-cheatsheet/blob/master/rst-cheatsheet.rst>`__.
+
+.. note::
+    Build the documentation locally with the following command:
+
+    .. code:: bash
+
+        $ conda env update -f ci/requirements-py36.yml
+        $ cd docs
+        $ make html
+
+    The built documentation should be available in the ``docs/build/``.
+
+If you need to add new functions to the API, run ``sphinx-autogen -o api api.rst`` from the
+``docs/source`` directory and add the functions to ``api.rst``.
+
+ .. _`pull requests`:
+ .. _pull-requests:
+
+Preparing Pull Requests
+-----------------------
+
+
+#. Fork the
+   `xskillscore GitHub repository <https://github.com/raybellwaves/xskillscore>`__.  It's
+   fine to use ``xskillscore`` as your fork repository name because it will live
+   under your user.
+
+#. Clone your fork locally using `git <https://git-scm.com/>`_, connect your repository
+   to the upstream (main project), and create a branch::
+
+    $ git clone git@github.com:YOUR_GITHUB_USERNAME/xskillscore.git
+    $ cd xskillscore
+    $ git remote add upstream git@github.com:raybellwaves/xskillscore.git
+
+    # now, to fix a bug or add feature create your own branch off "master":
+
+    $ git checkout -b your-bugfix-feature-branch-name master
+
+   If you need some help with Git, follow this quick start
+   guide: https://git.wiki.kernel.org/index.php/QuickStart
+
+#. Install dependencies into a new conda environment::
+
+    $ conda env update -f ci/requirements-py36.yml
+    $ conda activate xskillscore-dev
+
+#. Make an editable install of xskillscore by running::
+
+    $ pip install -e .
+
+#. Install `pre-commit <https://pre-commit.com>`_ and its hook on the ``xskillscore`` repo::
+
+     $ pip install --user pre-commit
+     $ pre-commit install
+
+   Afterwards ``pre-commit`` will run whenever you commit.
+
+   https://pre-commit.com/ is a framework for managing and maintaining multi-language pre-commit
+   hooks to ensure code-style and code formatting is consistent.
+
+    Now you have an environment called ``xskillscore-dev`` that you can work in.
+    You’ll need to make sure to activate that environment next time you want
+    to use it after closing the terminal or your system.
+
+    You can now edit your local working copy and run/add tests as necessary. Please follow
+    PEP-8 for naming. When committing, ``pre-commit`` will modify the files as needed, or
+    will generally be quite clear about what you need to do to pass the commit test.
+
+#. Break your edits up into reasonably sized commits::
+
+    $ git commit -a -m "<commit message>"
+    $ git push -u
+
+#. Run all the tests
+
+   Now running tests is as simple as issuing this command::
+
+    $ pytest xskillscore
+
+   Check that your contribution is covered by tests and therefore increases the overall test coverage::
+
+    $ coverage run --source xskillscore -m py.test
+    $ coverage report
+    $ coveralls
```

Comment:
Do we need to install anything for coveralls/put it in the yaml file?

Reply:
it gets installed in travis, but locally it also would be useful to have